.github/workflows/test-and-deploy.yml

```diff
@@ -0,0 +1,93 @@
+name: Test and Deploy
+on:
+  push:
+    branches: [ '*' ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+    # Run automatically at 8AM PST Monday-Friday
+    - cron: '0 15 * * 1-5'
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    strategy:
+      matrix:
+        ruby: [ 2.4, 2.5, 2.6, 2.7, 3.0, jruby-9.2 ]
+    steps:
+      - name: Checkout sendgrid-ruby
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+
+      - run: make install
+
+      - name: Set up linter
+        run: bundle add rubocop --version "~> 1.24.1" --group "development" --skip-install
+        if: ${{ matrix.ruby != '2.4' }}
+
+      - run: bundle install --with development
+
+      - name: Run linter
+        run: bundle exec rubocop
+        if: ${{ matrix.ruby != '2.4' }}
+
+      - name: Run tests
+        run: make test-docker
+
+  deploy:
+    name: Deploy
+    if: success() && github.ref_type == 'tag'
+    needs: [ test ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sendgrid-ruby
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.4
+          bundler-cache: true
+
+      - run: make install
+
+      - name: Publish to Rubygems
+        env:
+          GEM_HOST_API_KEY: ${{ secrets.RUBYGEMS_AUTH_TOKEN }}
+        run: |
+          mkdir -p $HOME/.gem
+          touch $HOME/.gem/credentials
+          chmod 0600 $HOME/.gem/credentials
+          printf -- "---\n:rubygems_api_key: ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
+          gem build *.gemspec
+          gem push *.gem
+
+  notify-on-failure:
+    name: Slack notify on failure
+    if: ${{ failure() && github.ref == 'refs/heads/main' && github.event_name != 'pull_request' }}
```

Comment:
Needs to test for tags as well. Should mirror this config for consistency: https://github.com/twilio/twilio-node/blob/871116ca2ab6be72294669d455c4db7273ca1496/.github/workflows/test-and-deploy.yml#L86-L102